session/session.go

```diff
@@ -559,9 +559,6 @@ func (s *session) retry(ctx context.Context, maxCnt uint) (err error) {
 		s.sessionVars.RetryInfo.ResetOffset()
 		for i, sr := range nh.history {
 			st := sr.st
-			if st.IsReadOnly() {
```

Comment:
Why removed the block?

Reply:
On https://github.com/pingcap/tidb/pull/9723/files#diff-5c9d0559245b0eecd9e72c2ec9562ac5R216
only if the statement is not a read-only one, it will be added to the history. So I think this check is redundant, all the statements in the history should not be read-only ones.